src/services/analytics.ts

```diff
@@ -0,0 +1,16 @@
+import { reportInteraction } from '@grafana/runtime';
+import pluginJson from '../plugin.json';
+
+// Helper function to create a unique interaction name for analytics
+const createInteractionName = (page: 'service_selection' | 'service_details', action: string) => {
+  return `grafana_${pluginJson.id}_${page}_${action}`;
```

Comment:
<3

Reply:
We should consider standarizing this across observability apps. Not sure how, but we LMT could benefit from it.